src/app/components/experiment/experiment.component.ts

```diff
@@ -172,12 +170,25 @@ export class ExperimentComponent implements OnInit, OnDestroy {
         });
     }
 
-    async changeDefaultTemplate(templateId: string | null) {
+    showSelectDefaultTemplateDialog() {
+        const dialogRef = this.dialog.open(ChooseTemplateComponent, {
+            minWidth: "20rem", maxWidth: "40rem", width: "60%", maxHeight: "95%",
+            data: this.uiTemplate
+        });
+        dialogRef.afterClosed().subscribe(templateId => {
+            if (templateId != null) {
+                const id = templateId === 'all-plugins' ? null : templateId;
+                this.updateExperimentDefaultTemplate(id);
+            }
+        });
+    }
+
+    async updateExperimentDefaultTemplate(templateId: string | null) {
         if (this.experimentId == null) {
+            console.warn("Experiment ID is null!");
```

Comment:
Should this show a warning?

Reply:
This method is only used in the experiment info tab, where the `experimentId` should never be `null`. Does it make sense to show a warning in that case?

Reply:
It is also used in the html template by a button which calls this with `null` if I have read the code correctly.

Reply:
If you actually do not allow `null`, then remove it from the type of `templateId` and let the typescript type checker do the verification for you.

But then you have no way to unset the template id. Currently the unset Button does call this method with `templateId=null`: https://github.com/UST-QuAntiL/qhana-ui/blob/d917a689e43c2e52f7387d23ef03077ee403afc5/src/app/components/experiment/experiment.component.html#L45